Subject: [PATCH] fix gcc-14 -Werror build

- pesigcheck.c: calloc(nmemb, size) argument order
- pesum.c: cast strrchr() result (infile is const char*)
--- a/src/pesigcheck.c	2023-01-31 17:55:13.000000000 +0300
+++ b/src/pesigcheck.c	2026-10-22 15:39:32.145824472 +0300
@@ -240,7 +240,7 @@
 
 	cert_iter iter;
 
-	reasonps = calloc(sizeof(struct reason), 512);
+	reasonps = calloc(512, sizeof(struct reason));
 	if (!reasonps)
 		err(1, "check_signature");
 
@@ -281,7 +281,7 @@
 
 			num_reasons += 16;
 
-			new_reasons = calloc(sizeof(struct reason), num_reasons);
+			new_reasons = calloc(num_reasons, sizeof(struct reason));
 			if (!new_reasons)
 				err(1, "check_signature");
 			reasonps = new_reasons;
--- a/src/pesum.c	2023-01-31 17:55:13.000000000 +0300
+++ b/src/pesum.c	2026-10-22 15:40:03.510480747 +0300
@@ -141,7 +141,7 @@
 	while ((infile = poptGetArg(optCon)) != NULL) {
 		pesign_context *ctxp = NULL;
 
-		char *ext = strrchr(infile, '.');
+		char *ext = (char *) strrchr(infile, '.');
 		if (ext && strcmp(ext, ".ko") == 0)
 			fmt = FORMAT_KERNEL_MODULE;
 
